src/DeadInstrumenter.cpp

```diff
@@ -258,25 +536,156 @@ AST_MATCHER(DoStmt, DoAndWhileNotMacroAndInMain) {
+            .bind("loop");
+    auto macroActions =
+        flatten(addDeleteMacroPre(changeTo(forBegin("loop"), cat("for"))),
+                insertAfter(LParenLoc("loop"), cat("\n\n#else\n{\n#endif\n\n")),
+                addDeleteMacro(SecondSemiForLoop("loop"), cat("\n\n")),
+                insertAfter(RParenLoc("loop"), cat("\n\n#endif\n\n")));
+    return applyFirst(
+        {makeRule(
+             compoundMatcher,
+             flatten(
+                 EmitDisableMacros ? edit(insertBefore(CStmtRBrace("body"),
+                                                       cat("\n\n#endif\n\n")))
+                                   : noEdits(),
+                 addMarkerBefore(statements("body"), cat("")),
+                 EmitDisableMacros
+                     ? edit(addDeleteMacro(CStmtLBrace("body"), cat("\n\n")))
+                     : noEdits(),
+                 EmitDisableMacros ? macroActions : noEdits())),
+         makeRule(
+             nonCompoundLoopMatcher,
+             flatten(addMarkerBefore(statementWithMacrosExpanded("body"),
+                                     cat("\n\n{\n\n")),
+                     EmitDisableMacros
+                         ? edit(addDeleteMacro(
+                               statementWithMacrosExpanded("body"), cat("")))
+                         : noEdits(),
+                     insertAfter(statementWithMacrosExpanded("body"),
+                                 EmitDisableMacros ? cat("\n\n#endif\n}\n\n")
+                                                   : cat("\n\n}\n\n")),
+                     EmitDisableMacros ? macroActions : noEdits()))});
+}
+
+RangeSelector whileBegin(std::string ID) {
+    return [ID](const clang::ast_matchers::MatchFinder::MatchResult &Result)
+               -> Expected<CharSourceRange> {
+        Expected<DynTypedNode> Node = getNode(Result.Nodes, ID);
+        if (!Node) {
+            llvm::outs() << "ERROR";
+            return Node.takeError();
+        }
+        const auto &SM = *Result.SourceManager;
+        return SM.getExpansionRange(
+            SourceRange(Node->get<WhileStmt>()->getWhileLoc()));
+    };
+}
+
+auto handleWhile() {
+    auto compoundMatcher =
+        whileStmt(inMainAndNotMacro,
+                  hasBody(compoundStmt(inMainAndNotMacro).bind("body")))
+            .bind("loop");
     auto nonCompoundLoopMatcher =
-        mapAnyOf(forStmt, whileStmt, cxxForRangeStmt)
-            .with(inMainAndNotMacro,
-                  hasBody(stmt(inMainAndNotMacro).bind("body")));
+        whileStmt(inMainAndNotMacro,
+                  hasBody(stmt(inMainAndNotMacro).bind("body")))
+            .bind("loop");
+    auto macroActions = flatten(
+        addDeleteMacroPre(changeTo(whileBegin("loop"), cat("while"))),
+        insertAfter(LParenLoc("loop"), cat("\n\n#endif\n\n")),
+        addDeleteMacro(RParenLoc("loop"), cat("\n\n")),
+        insertAfter(RParenLoc("loop"), cat("\n\n#else\n;\n#endif\n\n")));
```

Comment:
Potentially dangerous non-double `\n`? Or is this just problematic on the boundaries to "real code"?